addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1449,7 +1449,13 @@ registry.ImagesLazyLoading = publicWidget.Widget.extend({
         this._super(...arguments);
         const imgEls = this.$target[0].querySelectorAll('img[data-lazy-loading-initial-min-height]');
         for (const imgEl of imgEls) {
-            this._restoreImage(imgEl);
+            imgEl.style.minHeight = '1px';
+            wUtils.onceAllImagesLoaded($(imgEl)).then(() => {
+                if (this.isDestroyed()) {
+                    return;
+                }
+                this._restoreImage(imgEl);
+            });
```

Comment:
I do not understand why this change is needed here.

Reply:
@dmo-odoo  When we start editing the website if the image is not loaded properly then it will contain the `data-lazy-loading-initial-min-height` attribute that is removed from the ` _restoreImage` method. Here method ` _restoreImage` does not call for the specific images. If we don't stop the removal of that attribute from the images which is not loaded yet then at the discard time this small change is taken as the dom has changed.
Hence, the processing is done when the image is loaded successfully

Reply:
@Goaman Just fyi, another case here that can trigger unwanted changes.

Reply:
As it is an attribute, we can add a rule in `filterMutationRecords` of `wysiwyg.js` like:
`!(record.type === 'attributes' && record.name === 'data-lazy-loading-initial-min-height')`

Reply:
@Goaman I tried it but it's not working.

Reply:
Why did it not worked?

Reply:
@Goaman 
Here, when I click on the edit button 'data-lazy-loading-initial-min-height' is removed from the element, and in return because of that o_dirty class is added to that particular element. 

Even if I try your suggestion o_dirty class is still added to that Element and because of that, my other condition is not working correctly when I try to check if there is any o_dirty class available in dom.

Reply:
The method that add the `o_dirty` class should be `processRecords` of `_setObserver`. 
It only add the class on target of a mutation for the record that are not filtered.
My question is: if the record is properly filtered, what part of the code is adding the `o_dirty` class ?

Reply:
@Goaman  I have found the reason o_dirty class was added even after the record is properly filtered. the reason for that is that when 'data-lazy-loading-initial-min-height' is removed from the element at the same time min-height style is also removed from that element. I have added record.attributeName === 'style' in the above rule and it's working.
I have already push the changes in the branch.

Reply:
Hello @dhba-odoo, we have investigated the issue with @Goaman and found the root cause. The dirty observer was called before the public widgets were stopped and therefore it observed the changes caused by the destroying of the public widgets. I have updated the commit accordingly. @detrouxdev Could you validate our change here please ? In case there was a reason for calling `widgets_start_request` after `_setObserver` that we are not aware of ? cc @qsm-odoo

Reply:
Good catch 👍🏻  but I think that the issue might also be present in 15.0

Around this line (15.0):
https://github.com/odoo/odoo/blob/797f67bcb222142b79a9eee3e4a882ff679af9db/addons/website/static/src/js/menu/edit.js#L240

We pretty much replaced `_createWysiwyg` into an OWL `useEffect`, as the Wysiwyg is the main widget of this Component, it's started with the OWL lifecycle. When we did so, we moved what was done after `await this.wysiwyg.attachTo($('#wrapwrap'));` into this useEffect, while moving the code that set the observer into its own method, (I do not clearly remember why).

I just did a quick test and it seems like this could be the case on 15.0 as well:
- Drop a product snippet
- Save
- Enter edit mode
- Discard
- => Discard Popup when we should not have one (?)

So my opinion would be that I do not mind the fix, but it should probably be done in 15.0. And I am not familiar with what the OdooEditor observes during that time but maybe it should also not listen to changes during that time? You're probably better at judging that than me though ^^

Reply:
@detrouxdev Thank you for your input ! Since the changes in 15.0 would not forward port properly to 16.0 anyway, we'll be moving forward with this PR but I'm going to create a new task for fixing this issue in 15.0 as well with all the details you provided. Thanks !